packages/jest-snapshot/src/__tests__/mock_serializer.test.js

```diff
@@ -7,30 +7,122 @@
  */
 'use strict';
 
-const mock = jest.fn();
+const prettyFormat = require('pretty-format');
 
-afterEach(() => mock.mockReset());
+const plugin = require('../mock_serializer');
 
-test('empty with no calls mock', () => {
-  expect(mock).toMatchSnapshot();
+test('mock with 0 calls and default name', () => {
+  const fn = jest.fn();
+  expect(fn).toMatchSnapshot();
 });
 
-test('instantiated mock', () => {
+test('mock with 0 calls and default name in React element', () => {
+  const fn = jest.fn();
+  const val = {
+    $$typeof: Symbol.for('react.test.json'),
+    children: ['Mock me!'],
+    props: {
+      onClick: fn,
+    },
+    type: 'button',
+  };
+  expect(val).toMatchSnapshot();
+});
+
+test('mock with 1 calls and non-default name via new in object', () => {
+  const fn = jest.fn();
+  fn.mockName('MyConstructor');
   // eslint-disable-next-line no-new
-  new mock({name: 'some fine name'});
+  new fn({name: 'some fine name'});
+  const val = {
+    fn,
+  };
+  expect(val).toMatchSnapshot();
+});
 
-  expect(mock).toMatchSnapshot();
+test('mock with 1 calls in React element', () => {
+  const fn = jest.fn();
+  fn('Mocking you!');
+  const val = {
+    $$typeof: Symbol.for('react.test.json'),
+    children: ['Mock me!'],
+    props: {
+      onClick: fn,
+    },
+    type: 'button',
+  };
+  expect(val).toMatchSnapshot();
 });
 
-test('mock with calls', () => {
-  mock();
-  mock({foo: 'bar'}, 42);
+test('mock with 2 calls', () => {
+  const fn = jest.fn();
+  fn();
+  fn({foo: 'bar'}, 42);
+  expect(fn).toMatchSnapshot();
+});
 
-  expect(mock).toMatchSnapshot();
+test('indent option', () => {
+  const fn = jest.fn();
+  fn({key: 'value'});
+  const expected = [
+    '[MockFunction]',
+    'calls: Array [',
+    'Array [',
+    'Object {',
+    '"key": "value",',
+    '},',
+    '],',
+    ']',
+  ].join('\n');
+  expect(prettyFormat(fn, {indent: 0, plugins: [plugin]})).toBe(expected);
 });
 
-test('mock with name', () => {
-  const mockWithName = jest.fn().mockName('name of mock is nice');
+test('min option', () => {
+  const fn = jest.fn();
+  fn({key: 'value'});
+  const expected = '[MockFunction] calls: [[{"key": "value"}]]';
+  expect(prettyFormat(fn, {min: true, plugins: [plugin]})).toBe(expected);
+});
 
-  expect(mockWithName).toMatchSnapshot();
+test('maxDepth option', () => {
```

Comment:
Could you add a mock function with a name to a `toMatchSnapshot` assertion? I can see it working in this test, but a separate assertion would be awesome!

Reply:
See `MyConstructor` in `mock with 1 calls and non-default name via new in object 1`

Reply:
Yes, I can see it working, but it would be sweet with a test of just the name, no calls or wrapping objects